### PR TITLE
Add signal shifting optimization for feedforward operations

### PR DESCRIPTION
**Pull request by masa10-f**
## Summary

This PR introduces signal shifting optimization for feedforward operations in measurement-based quantum computing (MBQC). Signal shifting is a fundamental operation in the measurement calculus that transforms correction maps to enable maximum parallelization of measurement patterns. In this PR, I have implemented it in a feedforward module, decoupling from the pattern module.

  **Key changes:**
  - Added `signal_shifting()` function to the feedforward module
  - Added `propagate_correction_map()` as a subroutine.
  - Added comprehensive unit tests and integration tests
  - Updated documentation and changelog

  Testing

  Added comprehensive test coverage:
  - Plane-specific tests: Verified propagation behavior for XY, YZ, and XZ measurement
  planes
  - Edge case tests: Output node validation, optional zflow parameter handling
  - Integration tests: End-to-end testing with circuit compilation and simulation to
  verify correctness

  Documentation

  - Updated CHANGELOG.md with feature description
  - Added signal_shifting to API documentation in feedforward.rst
  - Enhanced docstrings with detailed parameter descriptions and return types

### CODE REVIEW

File: graphqomb/feedforward.py [hunk 1]
```diff
@@ -147,3 +153,127 @@ def check_flow(
...
+    -----
+    This function converts the correction maps into more parallel-friendly forms.
+    It is equivalent to the signal shifting technique in the measurement calculus.
+    """
+    if target_node in graph.output_node_indices:
+        msg = "Cannot propagate flow for output nodes."
+        raise ValueError(msg)
+
+    if zflow is None:
+        zflow = {node: odd_neighbors(xflow[node], graph) for node in xflow}
```
[the remark this answers is not in the document]
Comment (masa10-f) [reply graphqomb/feedforward.py]: Fixed in https://github.com/TeamGraphix/graphqomb/pull/130/commits/c93f52360d348d346f87c7db3689f3364dc184d0

File: graphqomb/feedforward.py [hunk 2]
```diff
@@ -147,3 +153,127 @@ def check_flow(
...
+    -------
+    `tuple`\[`dict`\[`int`, `set`\[`int`\]\], `dict`\[`int`, `set`\[`int`\]\]]
+        Updated correction maps for X and Z after signal shifting.
+    """
+    if zflow is None:
+        zflow = {node: odd_neighbors(xflow[node], graph) - {node} for node in xflow}
+
+    dag = dag_from_flow(graph, xflow, zflow)
+    topo_order = list(TopologicalSorter(dag).static_order())
+    topo_order.reverse()  # from parents to children
```
Comment (d1ssk) [inline graphqomb/feedforward.py]: from children to parents?

**Review by d1ssk — APPROVED** [review]
